consensus/istanbul/backend/handler.go

```diff
@@ -66,109 +58,70 @@ func (sb *Backend) HandleMsg(addr common.Address, msg p2p.Msg, peer consensus.Pe
 		}
 
 		if msg.Code == istanbul.DelegateSignMsg {
-			if sb.shouldHandleDelegateSign() {
+			if sb.shouldHandleDelegateSign(peer) {
 				go sb.delegateSignFeed.Send(istanbul.MessageEvent{Payload: data})
 				return true, nil
 			}
+			logger.Warn("Bad delegate sign message", "peer", peer)
```

Comment:
It seems like this should be logged at Error level. Also, I am not sure what to do if I saw that in my logs. Maybe something like "Received delegate sign request from unauthorized peer", would be a bit more clear.